src/App.tsx

```diff
@@ -15,48 +16,132 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortFieldValues {
+  default = '',
+  length = 'length',
+  abc = 'alphabetically',
+}
+
+interface ListObserverState {
+  sortType: SortFieldValues;
+  reverse: boolean;
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortType: SortFieldValues,
+  reverse: boolean,
+) {
+  let result = [...goods];
+
+  if (sortType) {
+    result.sort((good1, good2) => {
+      switch (true) {
+        case sortType === SortFieldValues.abc:
+          return good1.localeCompare(good2);
+
+        case sortType === SortFieldValues.length:
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    result = result.reverse();
+  }
+
+  return result;
+}
+
 export const App: React.FC = () => {
+  const [listObserver, setListObserver] = useState<ListObserverState>({
+    sortType: SortFieldValues.default,
+    reverse: false,
+  });
+
+  const preparedGoods = getPreparedGoods(
+    goodsFromServer,
+    listObserver.sortType,
+    listObserver.reverse,
+  );
+
+  const handleSort = (sortType: SortFieldValues) => {
+    setListObserver(prevState => ({
+      ...prevState,
+      sortType,
+    }));
+  };
+
+  const handleReverse = () => {
+    setListObserver(prevState => (
+      { ...prevState, reverse: !prevState.reverse }
+    ));
+  };
+
+  const handleReset = () => {
+    setListObserver({
+      sortType: SortFieldValues.default,
+      reverse: false,
+    });
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            { 'is-light': listObserver.sortType !== SortFieldValues.abc },
+          )}
+          onClick={() => handleSort(SortFieldValues.abc)}
         >
           Sort alphabetically
         </button>
 
         <button
           type="button"
-          className="button is-success is-light"
+          className={classNames(
+            'button is-success',
+            { 'is-light': listObserver.sortType !== SortFieldValues.length },
+          )}
+          onClick={() => handleSort(SortFieldValues.length)}
         >
           Sort by length
         </button>
 
         <button
           type="button"
-          className="button is-warning is-light"
+          className={classNames(
+            'button is-warning',
+            { 'is-light': !listObserver.reverse },
+          )}
+          onClick={handleReverse}
         >
           Reverse
         </button>
 
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
+        {(listObserver.sortType !== ''
+          || listObserver.reverse) && (
```

Comment:
better to make variable for this condition
```
const isResetButton = listObserver.sortType !== '' || listObserver.reverse;
```
```suggestion
        {isResetButton && (
```